### ISSUE TITLE
feat: add blanks-around-fences rule (MD031)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#185]
## Overview

Implement the \`blanks-around-fences\` rule (equivalent to markdownlint MD031 / remark-lint \`no-missing-blank-lines\`).

Closes part of #76.

## Problem

Fenced code blocks without surrounding blank lines can cause rendering issues in some Markdown processors and reduce readability.

**Invalid** (no blank line before or after the fence):

```
Some text
\`\`\`go
code here
\`\`\`
More text
```

**Valid** (blank lines around the fence):

```
Some text

\`\`\`go
code here
\`\`\`

More text
```

## Proposed Rule Key

\`blanks-around-fences\`

## Detection Logic

A violation is reported when a fenced code block (opening or closing fence) is not preceded or followed by a blank line, unless it is at the start or end of the file.

| Case | Valid? |
|---|---|
| Blank line before opening fence | ✅ |
| Blank line after closing fence | ✅ |
| Opening fence at start of file | ✅ |
| Closing fence at end of file | ✅ |
| Non-blank line immediately before opening fence | ❌ |
| Non-blank line immediately after closing fence | ❌ |
| Tilde fences (`~~~`) — same rules apply | ✅ / ❌ |

Both backtick (\`\`\`) and tilde (\`~~~\`) fences must be handled.

## Fix

No auto-fix. Report the line number of the fence with the missing blank line.

## References

- markdownlint MD031: https://github.com/DavidAnson/markdownlint/blob/main/doc/md031.md
- remark-lint \`no-missing-blank-lines\`
- #76 — master tracking issue (Priority 2)

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to create

- \`internal/rule/blanks_around_fences.go\` — \`CheckBlanksAroundFences\` function
- \`internal/rule/blanks_around_fences_test.go\` — unit tests (table-driven)
- \`e2e/fixtures/blanks_around_fences_valid.md\` — valid fixture
- \`e2e/fixtures/blanks_around_fences_violation.md\` — violation fixture
- \`e2e/fixtures/config-blanks-around-fences.json\` — e2e config

### Files to modify

- \`internal/config/config.go\` — register \`blanks-around-fences\` in \`DefaultConfigJSON\` and \`Default()\`
- \`internal/linter/linter.go\` — add \`blanks-around-fences\` to \`simpleRules\`
- \`e2e/e2e_test.go\` — add e2e test cases
- \`cmd/root_bench_test.go\` — add content exercising \`blanks-around-fences\` scan path without violations

### Checklist

- [ ] Implement \`CheckBlanksAroundFences(filename, lines, offset)\` in rule package
  - Detect opening fence with no blank line before it (unless at file start)
  - Detect closing fence with no blank line after it (unless at file end)
  - Handle both backtick (\`\`\`) and tilde (\`~~~\`) fences
  - Reuse existing fence tracking utilities (\`openingFenceMarker\`, \`IsClosingFence\`)
  - Skip violations inside nested fences (track open/close state)
- [ ] Write unit tests covering:
  - Blank line before and after fence (valid)
  - Missing blank before opening fence (invalid)
  - Missing blank after closing fence (invalid)
  - Fence at start of file (no preceding blank required)
  - Fence at end of file (no following blank required)
  - Tilde fences
  - Multiple fences in one file
  - Offset handling for frontmatter
- [ ] Register rule in config (\`DefaultConfigJSON\` + \`Default()\`)
- [ ] Add to \`simpleRules\` in linter
- [ ] Add e2e fixtures and test cases
- [ ] Update \`generateComplexMarkdown\` in \`cmd/root_bench_test.go\`
- [ ] Verify \`TestBenchmarkContentIsViolationFree\` passes
- [ ] \`make test-all\` passes
